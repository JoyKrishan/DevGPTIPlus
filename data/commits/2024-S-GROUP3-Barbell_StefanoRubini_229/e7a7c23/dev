!packages/db/src/scripts/python/newGen.py