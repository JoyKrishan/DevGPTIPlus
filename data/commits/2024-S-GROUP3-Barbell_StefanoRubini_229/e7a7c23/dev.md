Whenever you want to sync with the main branch, get the latest main branch and then merge it with your branch. This will help you to avoid merge conflicts. There is some funny tools can help you with this

- [ ] [Spotify API got HANDS](https://stackoverflow.com/questions/71631183/spotify-403-user-not-registered-in-the-developer-dashboard)
- [ ] [Spotify API got HANDS, same thing](https://www.reddit.com/r/learnprogramming/comments/11warb8/need_help_with_spotify_api/)
- [ ] [It seems the router didn't regconize our config](https://github.com/kristerkari/react-native-svg-transformer/issues/333)
- [ ] [Strange "warning" as mentioned above](https://www.reddit.com/r/expo/comments/1aif2w2/strange_error_missing_transformrouterroot_option/)
- [ ] [Prisma ew ew error](https://github.com/prisma/studio/issues/1195)
- Check the entire constants from expo-constants and see if it exist the hostUri
- If it doens't exist, you can add it manually.
```

### WebSocket

- [ ] [WebSocket](https://medium.com/@jonas.skackauskas/practical-implementation-of-websockets-with-next-js-and-trpc-eb5fb3b5c211)