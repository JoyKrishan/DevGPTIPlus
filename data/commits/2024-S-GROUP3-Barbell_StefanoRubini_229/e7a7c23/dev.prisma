enum Gender {
  MALE
  FEMALE
  PREFERNOTTOSAY
}

enum PosterSource {
  RAW
  GITHUB
}

model Post2 {
  id     String       @id @default(uuid())
  name   String
  text   String
  source PosterSource @default(RAW)

  // To return `Date`s intact through the API we need to add data transformers
  // https://trpc.io/docs/v11/data-transformers
  createdAt DateTime @unique @default(now())
  updatedAt DateTime @unique @default(now())
}

  published Boolean  @default(false)
  author    User?    @relation(fields: [authorId], references: [id])
  createdAt DateTime @default(now())
// * gender: User gender, default to prefer not to say.
  id                   Int               @id @unique @default(autoincrement())
  clerkId              String            @unique
  username             String            @unique @db.Char(20)
  streak               Int               @default(0)
  exerciseHistory      ExerciseLog[]
  savedWorkouts        WorkoutTemplate[]
  notifications        Notification[]    @relation("NotificationReceiver")
  sentMessages         Notification[]    @relation("NotificationSender")
  createdChats         Chat[]            @relation("ChatCreator")
  notificationsBanners Boolean           @default(true)
  gender               Gender?           @default(PREFERNOTTOSAY)
  finishedAt DateTime @default(now())
  duration   Int
  workoutTemplate   WorkoutTemplate @relation(fields: [workoutTemplateId], references: [id])
  workoutTemplateId Int
model WorkoutTemplate {
  id           Int          @id @unique @default(autoincrement())
  name         String       @db.Char(100)
  description  String?
  createdAt    DateTime     @default(now())
  updatedAt    DateTime     @updatedAt
  likes        Int          @default(0)
  exerciseIds  Int[]
  pastWorkouts WorkoutLog[]
  id           Int           @id @unique @default(autoincrement())
  name         String        @db.Char(100)
  note         String?
  bodyPart     BodyPart
  category     Category
  exerciseLogs ExerciseLog[]

  user   User? @relation(fields: [userId], references: [id])
  userId Int?
}

model ExerciseLog {
  id        Int      @id @unique @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  exercise   Exercise @relation(fields: [exerciseId], references: [id])
  exerciseId Int

  workoutTemplateId Int

  user   User @relation(fields: [userId], references: [id])
  userId Int
  id            String      @id @default(uuid())
  type          SetType     @default(NORMAL)
  unilateral    Boolean     @default(false)
  reps          Int[]
  weight        Float[]
  exerciseLog   ExerciseLog @relation(fields: [exerciseLogId], references: [id])
  exerciseLogId Int