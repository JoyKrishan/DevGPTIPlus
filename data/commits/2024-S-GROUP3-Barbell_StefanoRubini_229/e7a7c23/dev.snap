        "marginBottom": 20,
            "translateY": 1334,
        <Text
          className="p-6 text-center text-xl font-bold text-white"
          Select an Option
        </Text>
        <View>
          <Text>
            Child content goes here